questdb-rs/src/ingress/http.rs

```diff
@@ -383,3 +405,175 @@ pub(super) fn http_send_with_retries(
 
     retry_http_send(state, buf, request_timeout, retry_timeout, last_rep)
 }
+
+pub(super) fn get_line_protocol_version(
+    state: &HttpHandlerState,
+    settings_url: &str,
+) -> Result<(Option<Vec<LineProtocolVersion>>, LineProtocolVersion), Error> {
```

Comment:
Document return type: I would have expected a single `Vec<ProtocolVersion>`.

Reply:
Also ensure that there's tests for the following scenarios:
* The server returns 404.
  * Default to V1
* The server returns, but the field is missing from the JSON.
  * Default to V1
* The server returns only version 1 (if the v2 protocol gets toggled via config in the server).
* The server returns [1, 2].
  * Client uses V2.
* The server returns [2].
  * Client uses V2. 
* The server returns [MAX_SUPPORTED_VERSION + 1].
  * The client errors, since there's no compatible protocol.
  
For these tests, ensure create a buffer from `server.new_buffer()`, add a simple with just a timestamp and an f64 and flush it, checking the serialized behaviour.